robot_smach_states/src/robot_smach_states/navigation/find.py

```diff
@@ -94,8 +95,8 @@ class Find(smach.StateMachine):
     Find an entity based on a description. The description designator should
     have resolve type dict and it should contain at least a 'type' field
     """
-    def __init__(self, robot, knowledge, source_entity_designator, description_designator, area_name_designator,
-                 navigation_area_designator, found_entity_designator):
+    def __init__(self, robot, knowledge, source_entity_designator, description_designator, found_entity_designator,
+                 area_name="on_top_of", navigation_area_designator=""):
```

Comment:
I can understand the area_name default value, however the previous name suggests that it should be a designator and not a string, I assume you checked this? On top of that why would you give a default empty navigation_area_designator (again: designator?!)? I can't imagine that being a good idea in general.

Reply:
The previous names are deceptive as both area_name and navigation_area can be either strings of designators to strings.

The empty string is used to check whether this value was provided by the user much like you would use =None (which we could also use here). If it is empty we use a navigateToObserve

Reply:
> If it is empty we use a navigateToObserve
Could you please put this in the docstring as well? That would make it really clear. 
Kudo's for adding the docstring by the way, this makes everything better. Nice job!

Reply:
I think making it None is better for clarity.